### ISSUE TITLE
Tunnel Term Attributes validation condition for MP2P and MP2MP types

### ISSUE THREAD

**Issue by mholankar** [sonic-net/sonic-swss#2829]
For Tunnel Termination Tables - for MP2P type, we want to be able to provide the dst IP+ mask & src IP.
Also for MP2MP type, made sure dst IP and src IP attributes are allowed for the type

This is the Pull Request to make this fix. https://github.com/opencomputeproject/SAI/pull/1799.

However, this PR seems to cause a test failure in sairedis.

main::test_brcm_acl_limit: Fresh start
Killing syncd
Flushing redis
Starting syncd
Replay BCM56971B0/acl_limit.rec
terminate called after throwing an instance of 'std::runtime_error'
  what():  :- replay: failed to execute api: c: SAI_STATUS_MANDATORY_ATTRIBUTE_MISSING
player BCM56971B0/acl_limit.rec: exitcode: 134:
FAIL: BCM56971B0.pl
===================
1 of 5 tests failed
===================

It seems that for tunnel types SAI_TUNNEL_TERM_TABLE_ENTRY_TYPE_P2MP, the SAI_TUNNEL_TERM_TABLE_ENTRY_ATTR_SRC_IP attribute is missing for the test cases.

**Comment by mholankar** [comment]
@kcudnik opened this issue to track what we discussed

**Comment by kcudnik** [comment]
Just to clarify, that change in SAI will break backward compatibility, and it will require from now on that DST_IP and SRC_IP will be mandatory on create, which were not, this is causing test in sairedis repo to fail since those are attributes are missing now in test cases, this could be fixed easy by added them to test cases

but more important this will cause orchagent to fail when create tunel term table entry object, since those attributes are now required in swss/orchagent, and this needs to be updated

**Comment by kcudnik** [comment]
needed changes here: https://github.com/sonic-net/sonic-sairedis/pull/1255 for visibility

**Comment by prsunny** [comment]
@kcudnik , what is the ask here? any orchagent change expected?

**Comment by kcudnik** [comment]
Yes, OA needs to be updated after pooling this headers since IT will fail to create tunnel term table entry by missing new required attributes
